mock/environment/__init__.py

```diff
@@ -16,66 +16,120 @@
+                    'IQ_OIWFS', 'IQ_P2WFS', 'IQ_OIWFS_MEDIAN', 'IQ_P2WFS_MEDIAN', 'QAP_IQ', 'QAP_IQ_WFS_scaled',
+                    'Ratio_OIWFS', 'Ratio_P2WFS', 'QAP_IQ_WFS_scaled500', 'IQ_P2WFS_Zenith', 'IQ_OIWFS_Zenith',
+                    'IQ_P2WFS_MEDIAN_Zenith', 'QAP_IQ_WFS_scaled500_Zenith'})
+
+                self.site_data_by_night[site] = {}
+                local_site_data = input_data.iterrows()
+                for index, night in local_site_data:
+                    night_start_line = night
+                    night_date = night[Env._time_stamp].date()
+                    night_start_line["cc_band"] = Env._cc_band_to_float(night_start_line["cc_band"])
+                    night_start_line["iq_band"] /= 100
+                    logging.info(f'\tProccesing UTC night of {night_date}')
+                    night_list = [night_start_line]
+                    previous_line = night_start_line
+                    index2, current_line = next(local_site_data)
+                    current_line["cc_band"] = Env._cc_band_to_float(current_line["cc_band"])
+                    current_line["iq_band"] /= 100
+
+                    while current_line[Env._time_stamp] - previous_line[Env._time_stamp] < Env._day_difference:
+                        night_list.append(current_line)
+                        previous_line = current_line
+                        try:
+                            index3, current_line = next(local_site_data)
+                            current_line["cc_band"] = Env._cc_band_to_float(current_line["cc_band"])
+                            current_line["iq_band"] /= 100
+                        except StopIteration:
+                            logging.info("End of data")
+                            break
+                
+                    self.site_data_by_night[site][night_date] = night_list
+            
+                for night in self.site_data_by_night[site]:
+                    self.site_data_by_night[site][night] = pd.DataFrame(self.site_data_by_night[site][night])
+                    iq_band = self.site_data_by_night[site][night].iloc[0]["iq_band"]
+                    cc_band = self.site_data_by_night[site][night].iloc[0]["cc_band"]
+                    starting_index = self.site_data_by_night[site][night].index[0]
+
+                    if pd.isna(iq_band) and pd.isna(cc_band):
+                        self.site_data_by_night[site][night].at[starting_index, "iq_band"] = 1.0
+                        self.site_data_by_night[site][night].at[starting_index, "cc_band"] = 1.0
+                    elif pd.isna(iq_band) and (~pd.isna(cc_band)):
+                        self.site_data_by_night[site][night].at[starting_index, "iq_band"] = 1.0
+                    elif (~pd.isna(iq_band)) and pd.isna(cc_band):
+                        self.site_data_by_night[site][night].at[starting_index, "cc_band"] = 1.0
```

Comment:
I'm not sure you understand what `~` does in Python. It means to invert all the bits and doesn't mean `not`. In Python, you just use the word `not` for not.

This could be simplified quite a bit:
```
if pd.isna(iq_band):
    self.site_data_by_night[site][night].at[starting_index, "iq_band"] = 1.0

if pd.isna(cc_band):
    self.site_data_by_night[site][night].at[starting_index, "cc_band"] = 1.0
```

You don't need four cases. You can just check `iq_band` and `cc_band` separately and handle them.